### DISCUSSION TITLE
Switch Tabs

### DISCUSSION THREAD

**Discussion by JuanLH** [dbeaver/dbeaver#14662] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I am used to changing tabs with Ctrl + Tab or Ctrl + Shift + Tab to switch to previous tab.

**Describe the solution you'd like**
Set a shortcut to change open tabs. 
[Ctrl + Tab] for switch fordward
[Ctrl + Shift + Tab] for switch backward

**Comment by ShadelessFox** [comment, marked as answer]
Hello @JuanLH,

We do have shortcuts for that: <kbd>Ctrl</kbd>+<kbd>PageDown</kbd> and <kbd>Ctrl</kbd>+<kbd>PageUp</kbd>. 

You can rebind them to any other shortcuts in <kbd>Preferences</kbd> &rArr; <kbd>User Interface</kbd> &rArr; <kbd>Keys</kbd>.

**Reply by givihuda** [reply]
Please state what the names of the shortcuts are, not just their keyboard keys. Mac and Windows Ctrl doesn't necessarily map the same way.

**Reply by ShadelessFox** [reply]
Shortcuts are called `Next Tab` and `Previous Tab` accordingly.

**Reply by givihuda** [reply]
So this is wonderful, I'm having trouble finding the shortcut in Preferences -> User Interface -> Keys because DBeaver uses different labels for the shortcuts in documentation and app.

In Documentation, the name for Ctrl+PageUp is "Switch to the previous tab".

![image](https://user-images.githubusercontent.com/74785256/170078549-8cdd446b-c267-44cd-b404-5c5a25e697b1.png)

https://dbeaver.com/docs/wiki/Shortcuts/

In the app, trying to find this is an adventure.

![image](https://user-images.githubusercontent.com/74785256/170078675-2f3f5bfb-2585-4af9-987b-4829c0dc5e02.png)

![image](https://user-images.githubusercontent.com/74785256/170078724-e6f65723-f009-401a-a20c-570450eea3db.png)
![image](https://user-images.githubusercontent.com/74785256/170078765-800a0de5-3fd5-4627-8a40-282a415d7f3a.png)
![image](https://user-images.githubusercontent.com/74785256/170078825-f441fae7-9593-4542-a457-5b6fb88cf884.png)

**Reply by givihuda** [reply]
> Shortcuts are called `Next Tab` and `Previous Tab` accordingly.

I suppose these are correct?

![image](https://user-images.githubusercontent.com/74785256/170079393-e90d3d27-f45c-4acd-a8c5-83eaf333ae64.png)

They're labeled differently than docs... why? 

The shortcut for keys `⌃⇞` is labeled in docs as "Switch to the previous sub-tab", in app it's labeled "Next Tab". I'm very confused. How do I know this is actually going to do what it looks like? I guess I trial-and-error.

In fact Previous Tab has same shortcut in app:

![image](https://user-images.githubusercontent.com/74785256/170079820-942ee810-8307-4b4e-a896-005f6d7f3e59.png)

This is super-confusing.

When terminology remains the same in docs and app, the user isn't left to egg hunt or post on forums.

**Reply by givihuda** [reply]
I notice the two shortcuts actually have a differently-pointing arrow that looks the same until I look very closely; I'm actually not sure what those arrow characters refer to anyway. Tab? Shift-Tab?

**Reply by ShadelessFox** [reply]
You mean the <kbd>⇞</kbd> and <kbd>⇟</kbd> keys? I don't understand your question, sorry.

On MacBooks that lack these arrows you have to use <kbd>Fn</kbd>+<kbd>↑</kbd>, so shortcuts for switching tabs are <kbd>Fn</kbd>+<kbd>^</kbd>+<kbd>↑</kbd> and <kbd>Fn</kbd>+<kbd>^</kbd>+<kbd>↓</kbd>.

**Reply by givihuda** [reply]
> You mean the ⇞ and ⇟ keys? I don't understand your question, sorry.

Yes, those same symbols in the interface and that I've been mentioning...

Which keyboards have those arrows? I've never seen a keyboard with those arrows. I'm in the US. Are those arrows on your keyboards?

> I'm actually not sure what those arrow characters refer to anyway. Tab? Shift-Tab?

I didn't see you answer this.

**Reply by ShadelessFox** [reply]
I don't know which keyboards have them, perhaps some of the Magic Keyboards manufactured by Apple. I'm a Windows user, so I use a regular keyboard which doesn't have these keys.

In case of DBeaver (and Eclipse), these keys map to <kbd>Page Up</kbd> and <kbd>Page Down</kbd> respectively. This doesn't mean that they are the same and therefore interchangeable. They are just used as shortcut keys.

**Reply by givihuda** [reply]
If you're a collaborator, and the app and docs include symbols ⇞ and ⇟, and you're commenting on them, you should be familiar with them and know generally which sorts of keyboards they're on... I mean, why are they so prominently displayed in so many places if there's so much unfamiliarity with them?

Regardless, do you understand the point of my original replies? The app settings and documentation is confusing with these shortcuts/symbols, and needs to be cleaned up. The terminology is not consistent. It makes it hard to find things. Please see my original screenshots. As a collaborator, you should care to somehow have this cleaned up.

**Reply by ShadelessFox** [reply]
I agree with your statement regardless of our wiki's consistency. We need to update it.

By the way, it seems that [this keyboard](https://www.apple.com/uk/shop/product/MQ052B/A) has these keys.

**Reply by givihuda** [reply]
Thank you, however my statement is partly criticism of the Wiki's consistency, but it's not just the Wiki's inconsistency that's a problem, it's also the app's. These inconsistencies of course make it difficult to communicate about features here... that's what the primary developers should understand. I know as tech heads we focus on UI and documentation second to functionality, but with a product this useful, I think the terminology and documentation and settings should have as much importance since it leads to more usage of features and less questions here.

As far as your keyboard example, I don't see the same symbols on the keyboard as is in your docs or settings...

![image](https://user-images.githubusercontent.com/74785256/180066826-e5fb98a0-16ed-4e9f-8ee0-a8bef79196c7.png)

The Home/End buttons (matching my keyboard positions) have a similar-looking symbol, but so do the Page Up/Down keys... this is exactly the problem I'm talking about 😆 The DBeaver developers making a bunch of assumptions on what symbols refer to. I only remember ever seeing the ⇞ and ⇟ keys in relation to DBeaver... this is why _the terminology_, literally the _verbiage_, needs to be consistent... as in, the words for things, not just the symbols, because the symbols can mean different things across different languages, manufacturers, webpages, etc... typically at least the same _words_ mean the same things, and in the case of Page Up/Down, I'm pretty sure those have the same words and meanings across English variations.

**Reply by ornic** [reply]
> We do have shortcuts for that: Ctrl+PageDown and Ctrl+PageUp.

"Previous tab" and "Next tab" is not always the same as "move left" and "move right". For me, for example, previous/next is a matter of history of usage (as in Alt+Tab / Win+Tab in Windows). It would be very helpful to distinguish this situations.

**Reply by danceb** [reply]
You can achieve this with "next editor" and "previous editor". See: https://github.com/dbeaver/dbeaver/issues/1679#issuecomment-303488396

**Reply by ornic** [reply]
It's a pity that repeated pressing of the same hotkey don't move selected editor in popup window - you need other hand to make the selection. Otherwise it's the thing I was looking for, thank you!
